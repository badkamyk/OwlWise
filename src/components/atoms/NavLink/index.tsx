export default function NavLink({ href }: { href: string }) {
	return (
		<li>
			<a
				data-testid="nav-link"
				href={href === "Home" ? "/" : `/${href.toLowerCase()}`}
				className="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
				aria-current="page"
			>
				{href}
			</a>
		</li>
	);
}
