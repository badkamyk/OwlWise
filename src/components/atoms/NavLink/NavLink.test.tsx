import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import NavLink from "./index";

it("renders NavLink with props correctly", () => {
	const { container } = render(<NavLink href="Home" />);
	expect(
		container.getElementsByClassName(
			"block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
		)
	).toBeTruthy();
	expect(screen.getByRole("link")).toHaveProperty("href", "/");
	expect(container).toMatchSnapshot();
});
