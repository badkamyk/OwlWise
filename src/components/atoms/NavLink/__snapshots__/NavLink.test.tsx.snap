// Vitest Snapshot v1

exports[`renders NavLink with props correctly 1`] = `
<div>
  <li>
    <a
      aria-current="page"
      class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
      data-testid="nav-link"
      href="/"
    >
      Home
    </a>
  </li>
</div>
`;
